patches/chromium/os-17652_add_support_for_window_rotation.patch

```diff
@@ -0,0 +1,392 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tariq Bashir <120014322+t-bashir-bs@users.noreply.github.com>
+Date: Tue, 24 Sep 2024 13:44:25 +0100
+Subject: OS-17652: Add support for window rotation
+
+This patch introduces a new WebPreference and browser window API method
+setWindowTransform to control window rotation
+(90, 180, or 270 degrees) in BrightSign. The preference is set in
+WebContentsImpl and passed to RenderWidgetHostViewAura, which then
+applies the display transform hint on WindowTreeHost if rotation is
+requested. Method setWindowTransform also applies the display transform
+hint on WindowTreeHost if rotation is requested.
+Method WindowTreeHost::GetTransformedWindowBounds was added to
+compute the transformed bounds when Window::SetBoundsInternal and
+WindowTreeHost::UpdateCompositorScaleAndSize are invoked.
+This method ensures bounds are transposed only once by checking the aspect
+ratio of the current and new bounds, avoiding multiple transpositions.
```

Comment:
Does this work if width and height are equal? 1000x1000px for example?

Reply:
Yes, I tested landscape, portrait and square html widget rectangle sizes.